### ISSUE TITLE
Arbitrary angles and phases in ConvSimProcessor and issue reshaping to z=1

### ISSUE THREAD

**Issue by pptman** [andreabassi78/napari-sim-processor#11]
@andreabassi78, I have modified ConvSimProcessor to be able to cope with arbitrary number of angles and phases in a new branch "anglesandphases".  I have tested this with script for (3 angles, 3 phases) and with (1 angle, 3 phases) and this works fine.

However, I have found an issue when reshaping datasets with only one z-plane.  Reshaping seems to work, but the image appears black in the Napari display.  The data in the array still has the correct numerical values in its array as before and even the Napari contrast slider shows the correct min and max values until you hit reset and they switch to (0,1).  There is no problem when z>1 and reshaping with a=1 or p=1 works fine.  

This doesn't look like an issue with the ReShape widget, but might point to use doing something wrong with the way we use napari to visualise multi-dimensional data.

I am pretty confident, based on the (3, 3) and (1, 3) results that the changes to ConvSimProcessor work, but it is a bit difficult to test inside napari at the moment without being able to use z=1 datasets.

**Comment by andreabassi78** [comment]
Could not reproduce the problem on a random image stack :
data=np.random.random([1,3,1,50,50]) # even trying many different combinations of initial shapes
viewer.add_image(data)
... run reshape...
I'm wondering if the problem is given by the `reshaping order`.
Can you share a sample dataset that gives this issue?

**Comment by pptman** [comment]
I was using one of the fairSIM data sets, https://github.com/fairSIM/test-datasets/releases/download/SLM-SIM-Bielefeld/SLM-SIM_Tetraspeck200_680nm.tif .  Same problem on both mac and windows.

Before reshaping:
![image](https://user-images.githubusercontent.com/17027217/168688716-965f10c3-b05a-41fc-9950-001417a22702.png)


After reshaping:
![image](https://user-images.githubusercontent.com/17027217/168688623-e50b362c-cbb4-4150-b8cb-2218f63e9ff4.png)

**Comment by pptman** [comment]
I have found the problem.  If you change to z=1 when the z-slider is set to show a slice that no longer exists (z=2 for instance) then the display becomes invisible and the displayed z-plane can no longer be changed as the slider is not visible.  The solution is to reset all the slider values to zero after reshaping the array with:
`    viewer.dims.set_point(axis=[0,1,2], value=[0,0,0])
`
I will commit the changes once I have checked that the modifications to ConvSimProcessor work as expected.

**Comment by andreabassi78** [comment]
ok, I will doublecheck that the other images that appear during processing(spectrum,xcorr,sim) will not be affected by this change

**Comment by pptman** [comment]
The changes to ConvSimProcessor seem to work fine with the 4 angle, 3 phase bead dataset from fairSim.

I also noticed that the plot_eta circles were not being calculated correctly and have fixed this.  This was a problem with calculating the coherent cutoff, which actually should be 1 as k-space is normalised to make this 1.

I have added separate commits for both the above and for the slider change to the anglesandphases branch. 

I have also noticed a number of other small issues that I will continue to look at on this branch:

1. FilteredOSReconstruct returns a 2-D array if there is only one z-plane but scaling on the display expects a 3-D array.
2. Batch reconstruction with a single plane returns 2 planes, this is because batch reconstruction pads the stack to an even number of z-slices to make the FFT calculations easier.  I will check if this should be fixed.  Batch reconstruction is pretty meaningless with just one z-plane, but it would be good to be consistent with non-batch stack reconstruction when the number of z-planes is odd.
3. Reconstructing with 4 angles and 3 phases highlights the issue with the xcorr array being indexed according to phases rather than angles, as only 3 of the xcorr images are visible.
